crates/shared/src/price_estimation/competition/mod.rs

```diff
@@ -15,8 +16,16 @@ mod quote;
 
 /// Stage index and index within stage of an estimator stored in the
 /// [`CompetitionEstimator`] used as an identifier.
-#[derive(Copy, Debug, Clone, Default, Eq, PartialEq)]
-struct EstimatorIndex(usize, usize);
+#[derive(Copy, Debug, Clone, Eq, PartialEq)]
+struct EstimatorIndex(EstimatorStageIndex, usize);
+
+/// Stage index. It can be primary estimator (no index) or fallback estimator
+/// with its index
+#[derive(Copy, Debug, Clone, Eq, PartialEq)]
+enum EstimatorStageIndex {
```

Comment:
This is one of the things I don't fully like. But it is the best solution in order not to make a huge PR.